src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowHistoryV2Spec.groovy

```diff
@@ -298,6 +306,42 @@ class FlowHistoryV2Spec extends HealthCheckSpecification {
         ]
     }
 
+    @Tidy
+    @Tags(LOW_PRIORITY)
+    def "Info about a terminating switch is registered in flow history while rerouting"() {
+        given: "An active flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelperV2.randomFlow(srcSwitch, dstSwitch)
+        flowHelperV2.addFlow(flow)
+
+        when: "Deactivate the src switch"
+        def blockData = lockKeeper.knockoutSwitch(srcSwitch, RW)
+        def swIsActive = false
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwitch.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        then: "Flow goes DOWN"
```

Comment:
```
when: "Deactivate the src switch"
then: "Flow goes DOWN"
```
This currently looks like the fact of switch deactivation has brought the flow down, which is not true. Only ISL failure can change the flow status

Reply:
refactored
```
when: "Deactivate the src switch"
and: "Related ISLs are FAILED"
then: "Flow goes DOWN"
```